### PR TITLE
Use 'parliamentary group' (P4100) instead of 'political party' (P102)

### LINKED ISSUE #3: Use "parliamentary group" (P4100) instead of "political party" (P102)
There is the "WikiProject every politician" :
https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician

It uses "parliamentary group" (P4100) to bound the politician to its party in the parliament.

The bonus is that it will solve #2 ;-)

In a broader sense, I believe that it should be interesting to bring close the two queries. (e.g. the query for Landtag Nordrhein-Westfalen can be found there : https://www.wikidata.org/wiki/Wikidata:WikiProject_every_politician/Germany )

### PR DESCRIPTION
**Pull request by jonadem**
Fix #3

### CODE REVIEW

**Comment by k-nut** [comment]
Thanks a lot for your PR!

Unfortunately, it does not seem to work though. I just reenabled the deploy-previews so that now whenever you push changes, a new version of the site will be automatically deployed (see the checks in this PR and then click on Details in the deploy/netlify check).

If I check the deploy preview for the example given on the main page of the app, I only see two circles now instead of the 730 in the old version. https://deploy-preview-4--wikidata-parliament-svg.netlify.com/?parliament=Q1939555&term=Q15081430

Is this just because the data is not in Wikidata yet? And if so, is there maybe an example for where this works? :)

**Comment by jonadem** [comment]
It works with parliament=Q15705021&term=Q17173094 (current term of high federal chamber in Belgium).

**Comment by jonadem** [comment]
The difference is that : in my test, fractions has color, not in yours.
My query returns for example : "CDU/CSU Bundestag fraction" (Q1023134) as parliamentary group but this item doesn't have a color.

I don't know if this need to be added to wikidata or if the query need to be improved. What do you think ?

**Comment by k-nut** [comment]
I feel like it would be a valid addition to also add color to the parliamentary groups in Wikidata. The CDU/CSU group for example even officially list a color in their corporate design [here](http://cd.cducsu.de/de/basiselemente/farben.html)

**Comment by k-nut** [comment]
I just went ahead and added colours to the fractions. It seems to work quite well for the [current Bundestag](https://deploy-preview-4--wikidata-parliament-svg.netlify.com/?parliament=Q1939555&term=Q30579723) but unfortunately breaks all the old examples from NRW that are listed on the page. What would be your proposal for how to solve this?

**Comment by jonadem** [comment]
I don't think that all terms of NRW need to be listed in the default page.
We can agree on a specific number of the last ones (3 ?) and improve wikidata to make them work to merge the PR.
We will also need to refresh the term for the Bundestag in the land page.
Is this acceptable for you ?

**Comment by k-nut** [comment]
That is a really good point. Having three good examples for the last ones sounds like a great idea to me. Apart from the current term of the Bundestag, your Belgian example would also be great on the landing page to show that this is not just a German endeavour.

**Comment by jonadem** [comment]
OK. I will try to give a hand for NRW on wikidata with my little knowledge of the German world.

**Comment by k-nut** [comment]
Awesome, let me know if you get stuck somewhere.

**Comment by jonadem** [comment]
I have worked on NRW parliamentary term 14 and 15 going as far as I can.
Now, members of parliament without a parliamentary group are the one who are members of several political parties (and that will require some lookup).
This is the query that needs to be emptied for the 15th :
`SELECT ?item ?itemLabel ?group ?groupLabel ?party ?partyLabel WHERE {
  ?item p:P39 ?statement.
  ?statement ps:P39 wd:Q17781726.
  ?statement pq:P2937 wd:Q30544772 .
  OPTIONAL {?item wdt:P102 ?party.}
  MINUS { ?statement pq:P4100 ?group. }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],de". }
}
order by ?party`

**Comment by k-nut** [comment]
Nice! The [page listed as the source](https://www.landtag.nrw.de/portal/WWW/Webmaster/GB_I/I.1/Abgeordnete/Ehemalige_Abgeordnete/suche.jsp?suche_nach_vorname=&suche_nach_fraktion=alle&suche_nach_wahlkreis=&suche_nach_wahlperiode=Wahlperiode+15&aktuelleWP=17&suche_lebende_abgeordnete=alle&suchen=) for the membership properties (the official parliament page for this) actually lists the fractions that the members belonged to. It should be possible to extract that information from there as well. I'll see if I can get around to doing that over the holidays.

**Comment by k-nut** [comment]
I started taking another look at this and was wondering how to handle cases where politicians were part of the same parliament (such as Landtag NRW) but changed parties between terms.

Looking at [this politician](https://www.wikidata.org/wiki/Q2203730) we can see that he was a member for 4 terms. How would we add the fraction qualifier if he changed fractions in-between? Wouldn't this lead to the same problem that we currently have where one politician appears with multiple parties/fractions because the fraction qualifier itself does not point to a specific term? 
 
I feel like your proposed schema could solve this but I'm not sure how.

**Comment by jonadem** [comment]
It can be done like for this French politician : [Manuel Valls](https://www.wikidata.org/wiki/Q10287)
I.e. Add a "member of ..." for each time the parliamentary group change.

**Comment by k-nut** [comment]
Cool, that looks really good and future proof. Thanks for pointing me to it.

 I think I emptied the list that you provided by adding the respective qualifiers to the politicians' items. 

Do you want to change the examples that we link to on the index page and then we can merge this? 🚀

**Comment by jonadem** [comment]
Up to now, only NRW 14 and 15 are good to show. That's why I filtered out the others.

File: app/api.js [hunk 1]
```diff
@@ -40,6 +39,7 @@ SELECT ?period ?periodLabel ?number WHERE {
   ?period p:P31 ?thing .
   ?thing pq:P642 wd:Q571436.
   ?thing pq:P1545 ?number .
+  FILTER(xsd:integer(?number) >= 14 && xsd:integer(?number) <= 15).
```
Comment (k-nut) [inline app/api.js]: Maybe you can add a short comment above the function that explains why we filter for this range here? That might remind us to remove the filter at some point when the data has been added for the other terms as well.
After that, I'm super happy to merge this!

Comment (jonadem) [reply app/api.js]: Just in case you missed it, I updated my commit with your suggestion.

Comment (k-nut) [reply app/api.js]: Thanks for pointing this out. It seems like Github does not send notification emails for force pushes unfortunately.

**Comment by k-nut** [comment]
Thanks again, this looks like a much more future proof setup now! 🚀

**Pull request merged**